frontend/src/__tests__/api/authors.test.js

```diff
@@ -0,0 +1,124 @@
+import '@testing-library/jest-dom';
+import { act, renderHook, waitFor } from '@testing-library/react';
+import {
+  useAddAuthor,
+  useGetAuthor,
+  useGetAuthors,
+  allAuthorsQuery,
+  anAuthorQuery,
+  addAuthorMutation
+} from '../../../api/authors';
+import {
+  aBookQuery,
+  addBookMutation,
+  allBooksQuery,
+  removeBookMutation,
+  useAddBook,
+  useGetBook,
+  useGetBooks,
+  useRemoveBook
+} from '../../../api/books';
+import { MockedProvider } from '@apollo/client/testing';
+
+describe('api Authors test', () => {
+  it('should get all authors', async () => {
+    const getAuthorsMock = {
+      request: {
+        query: allAuthorsQuery,
+        variables: {}
+      },
+      result: {
+        data: {
+          getAuthors: [
+            {
+              id: '1',
+              firstName: 'J.K.',
+              lastName: 'Rowling'
+            },
+            {
+              id: '2',
+              firstName: 'Dan',
+              lastName: 'Gookin'
+            }
+          ]
+        }
+      }
+    };
+    const wrapper = ({ children }) => (
+      <MockedProvider mocks={[getAuthorsMock]} addTypename={false}>
+        {children}
+      </MockedProvider>
+    );
+
+    const { result } = renderHook(() => useGetAuthors(), { wrapper });
+    await waitFor(() => expect(result.current.authors.length).not.toBe(0));
+    expect(result.current.authorsError).toBeFalsy();
+    expect(result.current.authorsLoading).toBeFalsy();
+  });
+  it('should get an author', async () => {
+    const getAuthorMOCK = {
+      request: {
+        query: anAuthorQuery,
+        variables: { id: '1' }
+      },
+      result: {
+        data: {
+          getAuthor: {
+            id: '1',
+            firstName: 'J.K.',
+            lastName: 'Rowling',
+            books: [
+              {
+                id: '1',
+                title: 'Harry Potter'
+              }
+            ]
+          }
+        }
+      }
+    };
+    const wrapper = ({ children }) => (
+      <MockedProvider mocks={[getAuthorMOCK]} addTypename={false}>
+        {children}
+      </MockedProvider>
+    );
+
+    const { result } = renderHook(() => useGetAuthor('1'), { wrapper });
+    await waitFor(() => expect(result.current.authorLoading).toBeFalsy());
+    expect(JSON.stringify(result.current.author)).not.toBe('{}');
```

Comment:
Most the time, we want to check for what things *are* rather than what they *are not*. In this case, which one makes more sense?

Reply:
changed it to equal the expected return from author.